sql/modules/Reconciliation.sql

```diff
@@ -30,11 +30,17 @@ BEGIN;
 CREATE OR REPLACE FUNCTION reconciliation__submit_set(
         in_report_id int, in_line_ids int[]) RETURNS bool AS
 $$
+DECLARE
+        result NUMERIC;
 BEGIN
-        UPDATE cr_report set submitted = true where id = in_report_id;
-        PERFORM reconciliation__save_set(in_report_id, in_line_ids);
-
-        RETURN FOUND;
+        result = reconciliation__check_balanced(in_report_id);
+        IF result = 0 THEN
+            UPDATE cr_report set submitted = true where id = in_report_id;
+            PERFORM reconciliation__save_set(in_report_id, in_line_ids);
```

Comment:
If they have already been added, then what does the line above do?

Reply:
Let me rephrase: what guarantees are there that the `in_line_ids` is exactly the same set of records as there is currently already in the report?

Reply:
In other words: Is there any guarantee we're marking as reconciled, the same lines as the ones you used to check that the report is balanced?